### PR TITLE
Send `ipp_channel` metadata in payment intent from store management/POS use cases

### LINKED ISSUE #14478: [POS TPV Tracking] Send `ipp_channel` metadata in payment intent for IPP/POS
Technical plan: pdfdoF-5QB-p2

## iOS

- [x] Send `ipp_channel` metadata in payment intent for IPP/POS with value `mobile_store_management`/`mobile_pos` respectively https://github.com/woocommerce/woocommerce-ios/pull/14479

## [WooPayments plugin](https://github.com/Automattic/woocommerce-payments)

- [x] Display POS channel value based on the conclusion in pdfdoF-5UW-p2 https://github.com/Automattic/woocommerce-payments/pull/9802
- [ ] Exploration: set a flag when the plugin detects a POS transaction (might require server-side changes) pdfdoF-5UW-p2#comment-6983
- [ ] Exploration: support multiple selections for the channel filter pdfdoF-5UW-p2#comment-6984

## WooPayments server

- [x] Set a transaction's `channel` to `in_person_pos` if the IPP payment's metadata `ipp_channel` is `mobile_pos`

## Data engineering

- [x] Add the new `in_person_pos` channel value to the allowed values list of the channel column in `woocommerce_midput.wcpay_transaction_details` and `woocommerce_data.wcpay_transactions_cache`

### PR DESCRIPTION
**Pull request by jaclync**
Closes: #14478 

## Why

As we are launching POS to external merchants soon, we want to identify if a transaction is from POS or IPP in the mobile apps for data analysis purposes.

## How

As shared in pdfdoF-5QB-p2 > Apps, we want to add a new metadata `ipp_channel: mobile_store_management | mobile_pos` to the payment intent from IPP and POS respectively.

I thought about giving the channel parameter a default value of `.storeManagement`, but I'm afraid that this parameter will be forgotten and be the wrong value if we add more use cases in the future. I'm open to adding a default value if you think it will simplify the code though.

## Steps to reproduce

1. Create a new Jurassic Ninja site – include Jetpack, WooCommerce, WooPayments, and WooPayments Dev Tools
2. Set up Jetpack to link the site to WPCOM user in the app (or skip this if you're loggin in with site credentials)
3. Add a US address in WooCommerce settings
4. Activate WooPayments. Go through the onboarding steps in WP-Admin > Payments > Activate. Alternatively, you can go through the onboarding in the app
5. Launch the app and switch to your new site
6. Create a purchasable product
7. Go to the orders tab. Create an order and collect payment with a card reader --> the payment collection should succeed. after that, verify that the payment intent metadata includes `ipp_channel: mobile_store_management` in the Stripe account (* check instructions below)
8. Go to the menu tab > Point of Sale mode. Add product to the cart and pay for the order --> the payment collection should succeed. after that, verify that the payment intent metadata includes `ipp_channel: mobile_pos` in the Stripe account (* check instructions below)

### `*` How to check the transaction payment intent

Prerequisite: you have access to the WooPayments Stripe account (ask in `#payment-operations` if not)

- In wp-admin > WCPay Dev, tap `MC WCPay` to see the payments info for the site
- Under `Stripe account details` > Account ID, tap on the link to open the Stripe account (make sure you're logged in to WooPayments in Stripe, with test mode enabled)
- Tap Payments, find the payment that was just made
- On the payment details page, search for `ipp_channel` and verify the value is as expected

## Testing information

I also tested with local server-side changes.

## Screenshots

Screenshot 2024-11-21 at 2 03 28 PM

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added

### CODE REVIEW

File: Hardware/Hardware/CardReader/PaymentIntent.swift [hunk 1]
```diff
@@ -106,7 +111,8 @@ public extension PaymentIntent {
                           siteURL: String? = nil,
                           orderID: Int64? = nil,
                           orderKey: String? = nil,
-                          paymentType: PaymentTypes? = nil
+                          paymentType: PaymentTypes? = nil,
```
Comment (staskus) [inline Hardware/Hardware/CardReader/PaymentIntent.swift]: nit;

There's broken alignment for the rest of the parameters

**Review by staskus — APPROVED** [review]
✅  Thanks, works as described!

| POS | Store Management |
|-----|------------------|
| mobile_pos | mobile_store_management |

One note is that `ipp_channel` appears twice within payment details:

- charges -> data -> metadata -> ipp_channel
- metadata -> ipp_channel

------------

On a technical note, I was thinking if passing a channel all around could be avoided. If POS and Store Management would rely on different instances of dependencies, then we could inject the channel through an initializer. However, that's not an option now so I don't have good suggestions. Given this code rarely has a new entry point maybe thats a non-issue.

File: Hardware/Hardware/CardReader/PaymentIntent.swift (hunk 1 above)
Comment (jaclync) [reply Hardware/Hardware/CardReader/PaymentIntent.swift]: Updated the indentation of these misaligned parameters in https://github.com/woocommerce/woocommerce-ios/pull/14479/commits/b601a7c11a25b0dc01b4655393b1b0493fbd8939

**Comment by jaclync** [comment]
> On a technical note, I was thinking if passing a channel all around could be avoided. If POS and Store Management would rely on different instances of dependencies, then we could inject the channel through an initializer. However, that's not an option now so I don't have good suggestions. Given this code rarely has a new entry point maybe thats a non-issue.

Yea I agree that it's not pretty seeing a bunch of places where the channel is passed around. Initially, I tried DI'ing the channel to the initializers but decided to move it to the `collectPayment` function worrying that the dependencies could become a singleton in the app in the future. As the channel value isn't something we test normally, it might be worth making this parameter explicit in every payment collection use case.

**Comment by jaclync** [comment]
Merging this PR as it's approved to avoid merge conflicts, with the possibility of adding a feature flag or condition on the WooPayments version per decision pdfdoF-5UW-p2#comment-6983.

**Pull request merged**